src/many-ledger/src/storage.rs

```diff
@@ -812,18 +799,25 @@ impl LedgerStorage {
     }
 
     pub fn delete_account(&mut self, id: &Identity) -> Result<(), ManyError> {
-        self.persistent_store
-            .apply(&[(key_for_account(id), Op::Delete)])
-            .map_err(|e| ManyError::unknown(e.to_string()))?;
-
-        self.add_transaction(TransactionInfo::AccountDelete { account: *id });
+        let mut account = self
+            .get_account_even_disabled(id)
+            .ok_or_else(|| account::errors::unknown_account(*id))?;
+
+        if account.disabled.is_none() || account.disabled == Some(Either::Left(false)) {
+            account.disabled = Some(Either::Left(true));
+            self.commit_account(id, account)?;
+            self.add_transaction(TransactionInfo::AccountDelete { account: *id });
```

Comment:
The `AccountDelete` `TransactionInfo` should not be used anymore. It should be changed to `AccountDisabled`